### PR TITLE
refactor(flopy.utils.postprocessing.get_water_table)

### PR DESCRIPTION
**Pull request by aleaf**
* vectorize get_water_table
* remove per_idx argument. Instead, if a 3D head array is passed (nlay x nrow x ncol), a 2D water table is computed; if a 4D head array is passed (with multiple stress periods); a 3D water table is returned (nper x nrow x ncol). 
* Add valid_min and valid_max args to allow a valid range of heads to be specified, in addition to a specified 'nodata' value. This resolves an issue with the previous get_water_table function, which only had a single nodata arg, in situations where dry and inactive cells had separate nodata values.

### CODE REVIEW

File: flopy/utils/postprocessing.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def get_transmissivities(
     return T
 
 
-def get_water_table(heads, nodata, per_idx=None):
+def get_water_table(heads, nodata, valid_min=-1e-4, valid_max=3e4):
```
Comment (jdhughes-usgs) [inline flopy/utils/postprocessing.py:126]: `valid_min` and `valid_max` seem to be arbitrary values. Initial values of `None` might be better so filtering is only done if the user specifies values other than `None`

Comment (aleaf) [reply flopy/utils/postprocessing.py:126]: The goal with these was to set a default range of valid head values that would result in nodata values (outside this range) getting automatically masked without any additional user input. But maybe there's a smarter way to do this. In the MODFLOW 6 I/O instructions, it says that *Cells that have been assigned an IDOMAIN value of zero or less are assigned a head value of 1.0 x 10^30. Cells that have converted to dry are assigned a dry value of −1.0 x 10^30.* Would it make sense to instead have default arguments for ``dry_cell_value=-1e30`` and ``inactive_cell_value=1e30`` (which would correspond to ``hdry`` and ``hnoflo`` in MODFLOW 2005 language). Are there situations where additional no data values would need to be input or would that cover it?

**Comment by langevin-usgs** [comment]
Maybe there is a pattern in PlotUtilities._plot_array_helper() and/or PlotMapView.plot_array() with "masked_values" that could be adopted so that this function behaves similarly?  mbase has an interface to hdry and hnoflo for all model types, I believe, so that information is available.  Just a thought.

**Comment by jdhughes-usgs** [comment]
I like the `masked_values` option suggestion

**Comment by jdhughes-usgs** [comment]
@jlarsen-usgs could `hnoflo` and `hdry` be attributes in the model grid (may not be the appropriate place to store this data) then optionally `get_water_table()` could take a `modelgrid` and the masking could be done automatically. This could also be used in the plotting routines to automatically mask these values in the plots and avoid the need to supply `masked_values` for the standard case.

**Review by jdhughes-usgs — APPROVED** [review]

**Pull request merged**